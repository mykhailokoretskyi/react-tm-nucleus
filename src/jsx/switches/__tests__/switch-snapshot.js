import React from 'react';
import Switch from '../switch';
import ResaleSwitch from '../resaleSwitch';

import renderer from 'react-test-renderer';

describe('Switch', () => {
    it(' off matches snapshots', () => {
        const component = renderer.create(
            <Switch value={false} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' on matches snapshot', () => {
        const component = renderer.create(
            <Switch value={true} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ResaleSwitch', () => {
    it(' off matches snapshots', () => {
        const component = renderer.create(
            <ResaleSwitch value={false} >[+]</ResaleSwitch>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(' on matches snapshot', () => {
        const component = renderer.create(
            <ResaleSwitch value={true} >[+]</ResaleSwitch>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});