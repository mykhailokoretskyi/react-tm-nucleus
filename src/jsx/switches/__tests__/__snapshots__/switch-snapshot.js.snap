exports[`ResaleSwitch  off matches snapshots 1`] = `
<button
  className="toggle-switch__button"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  <div
    className="toggle-switch__indicator--resale toggle-switch__indicator--left">
    [+]
  </div>
</button>
`;

exports[`ResaleSwitch  on matches snapshot 1`] = `
<button
  className="toggle-switch__button toggle-switch__button--active-resale"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  <div
    className="toggle-switch__indicator--resale toggle-switch__indicator--right">
    [+]
  </div>
</button>
`;

exports[`Switch  off matches snapshots 1`] = `
<button
  className="toggle-switch__button"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  <div
    className="toggle-switch__indicator toggle-switch__indicator--left" />
</button>
`;

exports[`Switch  on matches snapshot 1`] = `
<button
  className="toggle-switch__button toggle-switch__button--active"
  onBlur={[Function bound _onBlur]}
  onClick={[Function bound _onClick]}
  onFocus={[Function bound _onFocus]}
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  <div
    className="toggle-switch__indicator toggle-switch__indicator--right" />
</button>
`;
